src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+    function isExistingBountyId(uint bountyId) public view returns (bool) {
+        return _bountyList.isExistingId(bountyId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function getClaimInformation(uint claimId)
+        external
+        view
+        validClaimId(claimId)
+        returns (Claim memory)
+    {
+        return _claimRegistry[claimId];
+    }
+
+    /// @inheritdoc IBountyManager
+    function listClaimIds() external view returns (uint[] memory) {
+        return _claimList.listIds();
+    }
+
+    /// @inheritdoc IBountyManager
+    function isExistingClaimId(uint claimId) public view returns (bool) {
+        return _claimList.isExistingId(claimId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function listClaimIdsForContributorAddress(address contributorAddrs)
+        external
+        view
+        returns (uint[] memory)
+    {
+        return contributorAddressToClaimIds[contributorAddrs].values();
+    }
+
+    //--------------------------------------------------------------------------
+    // Mutating Functions
+
+    /// @inheritdoc IBountyManager
+    function addBounty(
+        uint minimumPayoutAmount,
+        uint maximumPayoutAmount,
+        bytes calldata details
+    )
+        external
+        onlyRole(uint8(Roles.BountyAdmin))
+        validPayoutAmounts(minimumPayoutAmount, maximumPayoutAmount)
+        returns (uint id)
+    {
+        // Note ids start at 1.
+        uint bountyId = ++_nextId;
+
+        // Add Bounty id to the list.
+        _bountyList.addId(bountyId);
+
+        Bounty storage b = _bountyRegistry[bountyId];
+
+        b.minimumPayoutAmount = minimumPayoutAmount;
+        b.maximumPayoutAmount = maximumPayoutAmount;
+        b.details = details;
+
+        emit BountyAdded(
+            bountyId, minimumPayoutAmount, maximumPayoutAmount, details
+        );
+
+        return bountyId;
+    }
+
+    /// @inheritdoc IBountyManager
+    function updateBounty(uint bountyId, bytes calldata details)
+        external
+        onlyRole(uint8(Roles.BountyAdmin))
+        validBountyId(bountyId)
+    {
+        _bountyRegistry[bountyId].details = details;
+
+        emit BountyUpdated(bountyId, details);
+    }
+
+    /// @inheritdoc IBountyManager
+    function lockBounty(uint bountyId)
+        external
+        onlyRole(uint8(Roles.BountyAdmin))
+        validBountyId(bountyId)
+        notClaimed(bountyId)
+    {
+        _bountyRegistry[bountyId].claimedBy = type(uint).max;
+
+        emit BountyLocked(bountyId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function addClaim(
+        uint bountyId,
+        Contributor[] calldata contributors,
+        bytes calldata details
+    )
+        external
+        onlyRole(uint8(Roles.ClaimAdmin))
+        validBountyId(bountyId)
+        notClaimed(bountyId)
+        returns (uint id)
+    {
+        validContributorsForBounty(contributors, _bountyRegistry[bountyId]);
+        // Note ids start at 1.
```

Comment:
Not an accurate comment. Since `claimId` and `bountyId` both share the same variable: `_nextId`, all you can say is the IDs are non-zero, because it is impossible for both the IDs to start with 1.